lib/inventory/inventory.go

```diff
@@ -305,11 +310,19 @@ func (h *downstreamHandle) Sender() <-chan DownstreamSender {
 }
 
 func (h *downstreamHandle) CloseContext() context.Context {
-	return h.closeContext
+	return h.shutdownContext
 }
 
 func (h *downstreamHandle) Close() error {
-	h.cancel()
+	h.closeCancel()
+	return nil
+}
+
+func (h *downstreamHandle) Shutdown() error {
```

Comment:
nit: `Shutdown` should take a context, and should force-close and exit if the context is canceled.